src/plugins/walletConnectToDapps/components/DappRegistryGrid.tsx

```diff
@@ -0,0 +1,102 @@
+import { SearchIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Heading,
+  Image,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Link,
+  SimpleGrid,
+  Stack,
+  Text as PlainText,
+} from '@chakra-ui/react'
+import type { FC } from 'react'
+import { useEffect, useMemo } from 'react'
+import { useForm, useWatch } from 'react-hook-form'
+import { Text } from 'components/Text'
+
+import type { RegistryItem } from '../types'
+import { PageInput } from './PageInput'
+
+const registryItems: RegistryItem[] = require('../registry.json')
+
+const PAGE_SIZE = 20
+
+export const DappRegistryGrid: FC = () => {
+  const { register, setValue, control } = useForm<{ search: string; page: number }>({
+    mode: 'onChange',
+    defaultValues: { search: '', page: 0 },
+  })
+
+  const search = useWatch({ control, name: 'search' })
+  const page = useWatch({ control, name: 'page' })
+  useEffect(() => setValue('page', 0), [search, setValue])
+
+  const filteredListings = useMemo(
+    () =>
+      registryItems.filter(
+        registryItem => !search || registryItem.name.toLowerCase().includes(search.toLowerCase()),
+      ),
+    [search],
+  )
+
+  const maxPage = Math.floor(filteredListings.length / PAGE_SIZE)
+
+  return (
+    <Box>
+      <Stack direction='row' alignItems='center' mb={4}>
+        <Heading flex={1} fontSize='2xl'>
+          <Text translation='plugins.walletConnectToDapps.availableDapps' />
+        </Heading>
+        <Box>
+          <InputGroup>
+            <InputLeftElement pointerEvents='none'>
+              <SearchIcon color='gray.700' />
+            </InputLeftElement>
+            <Input
+              {...register('search')}
+              autoComplete='off'
+              type='text'
+              placeholder='Search'
+              pl={10}
+              variant='filled'
+            />
+          </InputGroup>
+        </Box>
+        <PageInput value={page} max={maxPage} onChange={value => setValue('page', value)} />
+      </Stack>
+      <SimpleGrid columns={{ lg: 4, sm: 2, base: 1 }} spacing={4}>
+        {filteredListings.slice(page * PAGE_SIZE, (page + 1) * PAGE_SIZE).map(listing => (
```

Comment:
@reallybeard we want some sort of empty state here too

<img width="1041" alt="image" src="https://user-images.githubusercontent.com/88504456/190429880-65629b36-972e-4410-86a9-ae799e1ecc49.png">

Reply:
![Screen Shot 2022-09-20 at 4 11 26 PM](https://user-images.githubusercontent.com/89934888/191365348-0d34cb6b-8425-4bf8-b9e4-c5f26378d7fc.png)